public/app/plugins/panel/geomap/layers/data/markersLayer.tsx

```diff
@@ -132,11 +111,50 @@ export const markersLayer: MapLayerRegistryItem<MarkersConfig> = {
               styleConfig: style,
               size: style.dims?.size,
               layerName: options.name,
-              layer: vectorLayer,
+              layer: symbolLayer,
             });
           }
 
           source.update(frame);
+          source.forEachFeature((feature) => {
+            const idx: number = feature.get('rowIndex');
+            const dims = style.dims;
+            const values = { ...style.base };
+
+            if (dims?.color) {
+              values.color = dims.color.get(idx);
+            }
+            if (dims?.size) {
+              values.size = dims.size.get(idx);
+            }
+            if (dims?.text) {
+              values.text = dims.text.get(idx);
+            }
+            if (dims?.rotation) {
+              values.rotation = dims.rotation.get(idx);
+            }
+            const colorString = tinycolor(theme.visualization.getColorByName(values.color)).toString();
+            const colorValues = getRGBValues(colorString);
+
+            const radius = values.size ?? DEFAULT_SIZE;
+            const displacement = getDisplacement(values.symbolAlign ?? defaultStyleConfig.symbolAlign, radius);
+
+            // WebGLPointsLayer uses style expressions instead of style functions
+            feature.setProperties({ red: colorValues?.r ?? 255 });
+            feature.setProperties({ green: colorValues?.g ?? 255 });
+            feature.setProperties({ blue: colorValues?.b ?? 255 });
+            feature.setProperties({ size: (values.size ?? 1) * 2 });
```

Comment:
This ` * 2` is causing the discrepancy. I added this early on because circle markers were not matching the existing size. As it turns out, the old implementation's circle markers are too large (the red square is how big the marker should be, 25px):

![image](https://github.com/user-attachments/assets/700ff6be-d173-4fd1-8e34-9f6535b0f1a7)

After removing this ` * 2`, this is what the markers look like in WebGL:

![image](https://github.com/user-attachments/assets/9719bc31-c6d6-4b45-9c67-d4cdc0fc353c)

What do you think @adela-almasan?